### PR TITLE
Update MP metrics doc

### LINKED ISSUE #4198: [Doc:] Helidon MP Metrics
Helidon MP Metrics documentation (according to the new documentation structure).

Path: `/HelidonMP/Observability`

- [x] Overview (may be reused from current doc)
- [x] Maven Dependency (may be reused from current doc)
- [x] Usage (may be reused from Metrics Spec doc)
- [x] API (may be reused from Metrics Spec doc/website)
- [x] Configuration (should be created)
- [x] Reference (generated)
- [x] Examples (may be reused from Helidon repo)
- [x] Additional Information (should include `Metric-Capable Modules`, `Metrics for Exemplars` and `Micrometer`)

### PR DESCRIPTION
**Pull request by tjquinno**
See #4198

Mitia originally suggested moving Micrometer support, exemplar support, and metrics-capable modules to Additional Information on the main metrics page, which I know is our general tendency in the doc redesign.

IMO that made it much too long. At least for the moment I've kept those areas as separate pages which follow the new outline.

BTW, many of the changes affect shared files used by both MP and SE, so with luck the SE metrics update will go more quickly than it would have otherwise.

### CODE REVIEW

**Review by spericas — APPROVED** [review]
Mostly focused on the content. The new org is a bit difficult to follow without building the site, I guess.

**Review by spericas — APPROVED** [review]

File: docs/includes/metrics/micrometer-shared.adoc [hunk 1]
```diff
@@ -90,22 +87,39 @@ For example, Helidon configures the built-in Prometheus registry using `Promethe
...
+ifdef::mp-flavor[]
+[source,properties]
+----
+micrometer.builtin-registries.0.type=prometheus
+micrometer.builtin-registries.0.prefix=myPrefix
+----
+endif::[]
 Note that the first config example is equivalent to the default Helidon Micrometer behavior; Helidon by default supports the Prometheus meter registry.
 
 The configuration keys that are valid for the `builtin-registries` child entries depend on the type of Micrometer meter
```
Comment (ljamen) [inline docs/includes/metrics/micrometer-shared.adoc:125]: provide a link to the documentation?

File: docs/includes/metrics/prometheus-exemplar-support.adoc [hunk 2]
```diff
@@ -37,67 +60,100 @@ Declare the following dependency in your project:
 </dependency>
 ----
 
+Also, include the Helidon integration module for a tracing implementation...for example
```
Comment (ljamen) [inline docs/includes/metrics/prometheus-exemplar-support.adoc:63]: Does the ellipsis indicate something is missing and will be added?

File: docs/mp/metrics/micrometer.adoc [hunk 3]
```diff
@@ -16,18 +16,56 @@
...
+In addition to annotating your methods, your code
+include::{rootdir}/includes/metrics/micrometer-shared.adoc[tag=use-micrometer-api]
+
+
+include::{rootdir}/includes/metrics/micrometer-shared.adoc[tag=accessing-endpoint-intro]
+
+
+
+
+== API
```
Comment (ljamen) [inline docs/mp/metrics/micrometer.adoc:52]: provide some context about available APIs (options, usage?) We typically try to include some text in the H1/H2 sections. Not required but sometimes easier to navigate.

**Review by ljamen — COMMENTED** [review]
changes look good - a few small comments.

File: docs/includes/metrics/metrics-capable-components.adoc [hunk 4]
```diff
@@ -55,10 +67,12 @@ endif::[]
 
 As you plan and write Helidon components and applications,
 you make some choices about exactly how your code will use metrics.
-This guide gives some background information,
+This document gives some background information,
 describes the choices you face, explains their ramifications, and provides some code examples.
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/metrics/metrics-capable-components.adoc:71]: Fixed

File: docs/includes/metrics/metrics-capable-components.adoc [hunk 5]
```diff
@@ -55,10 +67,12 @@ endif::[]
...
 
-== Categorizing Metrics Usage
+== Usage
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/metrics/metrics-capable-components.adoc:73]: Fixed

File: docs/includes/metrics/metrics-capable-components.adoc [hunk 6]
```diff
@@ -82,7 +96,7 @@ The type of module dictates the compile-time dependency you declare in the proje
 
 Whenever possible, if your component or app uses metrics write it as metrics-capable code.
 
-== Understanding the Two Metrics Implementations
+=== Understanding the Two Metrics Implementations
 Helidon provides two metrics implementations.
 
 * _Full-featured_ metrics allows registering, removing, and updating metrics and observing metrics' changing values.
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/metrics/metrics-capable-components.adoc:102]: Fixed

File: docs/includes/metrics/metrics-config.adoc [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+* one matching metric names to _include_.
+
+Helidon updates and reports a metric only if two conditions hold:
+
+* the metric name _does not_ match the `exclude` regex pattern (if you define one), and
+* either
+** there is no `include` regex pattern, or
+** the metric name matches the `include` pattern.
+
+[CAUTION]
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/metrics/metrics-config.adoc:82]: Changed to a NOTE

File: docs/includes/metrics/prometheus-exemplar-support.adoc (hunk 2 above)
Comment (tjquinno) [reply docs/includes/metrics/prometheus-exemplar-support.adoc:63]: It looks as if this was already fixed in an earlier revision. Now reads "Also, include the Helidon integration module for a tracing implementation (such as" and then the SE or MP `ifdef` supplies the rest of the sentence.

File: docs/mp/metrics/micrometer.adoc (hunk 3 above)
Comment (tjquinno) [reply docs/mp/metrics/micrometer.adoc:52]: Fixed

File: docs/includes/metrics/micrometer-shared.adoc (hunk 1 above)
Comment (tjquinno) [reply docs/includes/metrics/micrometer-shared.adoc:125]: Added a link to the Micrometer Prometheus registry config interface.

File: docs/includes/metrics/metrics-capable-components.adoc [hunk 8]
```diff
@@ -256,12 +275,15 @@ Helidon returns either a full-featured `RegistryFactory` or a minimal one, depen
...
 
+// end::writing-component[]
+
 // end::writing-code-beginning[]
 
 
 // tag::writing-code-ending[]
 
-==== An Example: Docker Images
+== Examples
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/metrics/metrics-capable-components.adoc:285]: Probably not. I now know it's discouraged to have a single header at a given level. I've promoted the content for the Docker Images example to just under the H2.

File: docs/includes/metrics/metrics-capable-components.adoc [hunk 9]
```diff
@@ -82,7 +96,7 @@ The type of module dictates the compile-time dependency you declare in the proje
 
 Whenever possible, if your component or app uses metrics write it as metrics-capable code.
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/metrics/metrics-capable-components.adoc:97]: Fixed.

**Pull request merged**